azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -281,6 +312,23 @@ func expandKustoClusterSku(d *schema.ResourceData) (*kusto.AzureSku, error) {
 	return azureSku, nil
 }
 
+func expandKustoClusterVNET(input []interface{}) *kusto.VirtualNetworkConfiguration {
+	if len(input) == 0 {
```

Comment:
Just in case, we'll also want to check that `input[0] == nil` here too 

```suggestion
	if len(input) == 0 || input[0] == nil {
```